libs/smart-ngrx/src/actions/action.service/load-by-ids.class.ts

```diff
@@ -0,0 +1,129 @@
+import { Dictionary } from '@ngrx/entity';
+import { Store } from '@ngrx/store';
+import { map, Observable, Subject, take, withLatestFrom } from 'rxjs';
+
+import { mergeRowsWithEntities } from '../../common/merge-rows-with-entities.function';
+import { registerEntityRows } from '../../mark-and-delete/register-entity-rows.function';
+import { defaultRows } from '../../reducers/default-rows.function';
+import { SmartNgRXRowBase } from '../../types/smart-ngrx-row-base.interface';
+import { ActionGroup } from '../action-group.interface';
+import { bufferIdsAction } from './buffer-ids-action.function';
+
+function notAPreloadId(c: string): boolean {
+  return !['index-', 'indexNoOp-'].some((v) => c.startsWith(v));
+}
+
+/**
+ * Manages the loading of rows by their Ids
+ */
+export class LoadByIds {
+  private defaultRow!: (id: string) => SmartNgRXRowBase;
+  private actions!: ActionGroup;
+  private loadByIdsSubject = new Subject<string[]>();
+  private entities!: Observable<Dictionary<SmartNgRXRowBase>>;
+  /**
+   * The constructor for the LoadByIds class.
+   *
+   * @param feature the name of the feature this class is for
+   * @param entity the name of the entity this class is for
+   * @param store the store to dispatch the actions to
+   */
+  constructor(
+    private feature: string,
+    private entity: string,
+    private store: Store,
+  ) {}
+
+  /**
+   * Initializes the service with the actions and starts the dispatcher.
+   *
+   * @param actions the actions to use
+   * @param entities the entities to check for loading
+   * @param defaultRow the default row to use
+   */
+  init(
+    actions: ActionGroup,
+    entities: Observable<Dictionary<SmartNgRXRowBase>>,
+    defaultRow: (id: string) => SmartNgRXRowBase,
+  ): void {
+    this.actions = actions;
+    this.entities = entities;
+    this.defaultRow = defaultRow;
+    this.loadByIdsDispatcher();
+  }
```

Comment:
The reason we use an init method here is because we don't want to call this.loadByIdsDispatcher() as part of the constructor.  running executable code during a constructor is a bad idea. Constructors should ONLY initialize variables.

While we could move the 3 lines of initialization into the constructor, they are strongly tied to the call to loadByIdsDispatcher() so we will leave them there